broker/fragment/store_gcs.go

```diff
@@ -167,7 +173,18 @@ func (s *gcsBackend) gcsClient(ep *url.URL) (cfg GSStoreConfig, client *storage.
 	creds, err := google.FindDefaultCredentials(ctx, storage.ScopeFullControl)
 	if err != nil {
 		return
-	} else if creds.JSON != nil {
+	}
+
+	// best effort to determine if JWT credentials are for external account
+	externalAccount := false
+	if creds.JSON != nil {
+		var f credentialsFile
+		if err := json.Unmarshal(creds.JSON, &f); err == nil {
+			externalAccount = f.Type == "external_account"
+		}
+	}
+
+	if creds.JSON != nil && !externalAccount {
```

Comment:
what if it is an external account?

Reply:
With an external_account we now fall into the else portion which the client is prepared to work with a number of formats including external_account. It was a case that when finding the default credentials for external_account format they still filled in creds.JSON which we did not really need and made us fall into the signed url code.